.nix/flake.nix

```diff
@@ -86,29 +84,31 @@
         rustGPUEnv = import ./rust-gpu.nix { inherit pkgs; };
 
         libPath = "${pkgs.lib.makeLibraryPath buildInputs}:${cefEnv.CEF_PATH}";
-      in
-      {
+      in {
         devShells.default = pkgs.mkShell ({
           packages = buildInputs ++ buildTools ++ devTools;
 
           LD_LIBRARY_PATH = libPath;
-          XDG_DATA_DIRS = "${pkgs.gsettings-desktop-schemas}/share/gsettings-schemas/${pkgs.gsettings-desktop-schemas.name}:${pkgs.gtk3}/share/gsettings-schemas/${pkgs.gtk3.name}:$XDG_DATA_DIRS";
+          XDG_DATA_DIRS =
+            "${pkgs.gsettings-desktop-schemas}/share/gsettings-schemas/${pkgs.gsettings-desktop-schemas.name}:${pkgs.gtk3}/share/gsettings-schemas/${pkgs.gtk3.name}:$XDG_DATA_DIRS";
 
           shellHook = ''
             alias cargo='mold --run cargo'
           '';
         } // cefEnv // rustGPUEnv);
 
-        packages.default = pkgs.stdenv.mkDerivation(finalAttrs: {
+        packages.default = pkgs.rustPlatform.buildRustPackage (finalAttrs: {
           pname = "graphite-editor";
           version = "unstable";
           src = pkgs.lib.cleanSource ./..;
 
-          cargoDeps = pkgs.rustPlatform.fetchCargoVendor {
-            src = finalAttrs.src;
-            hash = "sha256-BVIQIZbGW19Rof0J7U2r6XFCUC52hb7+uaE1di4bV4A=";
+          cargoLock = {
+            lockFile = ../Cargo.lock;
+            allowBuiltinFetchGit = true;
           };
 
+          
+          # TODO: Remove the need for this hash by getting individual package hashes from package-lock.json. @timon-schelling and @bluelinden tried this on 2025-10-17, but we were unsuccessful due to the bad documentation around pkgs.importNpmLock. We hope to try this again sometime.
```

Comment:
I would prefer just: `# TODO: Remove the need for this hash by relying on package-lock.json`